src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/References/AssemblyReferenceHandler.cs

```diff
@@ -16,25 +16,45 @@ internal class AssemblyReferenceHandler : AbstractReferenceHandler
         { }
 
         protected override Task RemoveReferenceAsync(ConfiguredProjectServices services,
-            ProjectSystemReferenceInfo referencesInfo)
+            string itemSpecification)
         {
             Assumes.Present(services.AssemblyReferences);
 
             AssemblyName? assemblyName = null;
             string? assemblyPath = null;
 
-            if (Path.IsPathRooted((referencesInfo.ItemSpecification)))
+            if (Path.IsPathRooted(itemSpecification))
             {
-                assemblyPath = referencesInfo.ItemSpecification;
+                assemblyPath = itemSpecification;
             }
             else
             {
-                assemblyName = new AssemblyName(referencesInfo.ItemSpecification);
+                assemblyName = new AssemblyName(itemSpecification);
             }
 
             return services.AssemblyReferences.RemoveAsync(assemblyName, assemblyPath);
         }
 
+        protected override Task AddReferenceAsync(ConfiguredProjectServices services, string itemSpecification)
+        {
+            Assumes.Present(services.AssemblyReferences);
+
+            AssemblyName? assemblyName = null;
+            string? assemblyPath = null;
+
+            if (Path.IsPathRooted(itemSpecification))
+            {
+                assemblyPath = itemSpecification;
+            }
+            else
+            {
+                assemblyName = new AssemblyName(itemSpecification);
+            }
```

Comment:
This refactoring won't work. The blocks assign different variables.

Reply:
Derp. That was a mis-read on my part. It would be:
```suggestion
            bool isPathRooted = Path.IsPathRooted(itemSpecification);
            string? assemblyName = isPathRooted ? null : new AssemblyName(itemSpecification);
            string? assemblyPath = isPathRooted ? itemSpecification : null;
```